### DISCUSSION TITLE
Upgrading from v5 to v6 causing native-base to not work.

### DISCUSSION THREAD

**Discussion by SethY8s** [storybookjs/react-native#394] [Q&A] [answered]
Hey everyone, so recently at work we decided to start using storybook and we initially started with react-native v5 before we found out that maybe using the beta version of v6 would be better. When I tried to convert and just play around with v6, I found out that whenever I used native-base components my app would crash and I would get an error saying "Can't find variable: HTMLElement". This error would be in several node modules relating to react-native, and I was wondering if anyone knew a workaround we could still use native-base and @storybook/react-native beta 6. 

Things I have tried to fix:
-Delete node_modules and reinstall them.
-reinstalling different versions of storybook

I highly recommend checking out my repo [here](https://github.com/maetio/expo-template/tree/storybook) since you can see all my code in .storybook (note that the folder called storybook is just old v5 code). Another important thing to note is that I know react-native storybook v6 is indeed working since I can make normal components with just react-native elements such as View and Text components. I have been looking hard to fix this all day and would really appreciate it if someone had any insight on how to fix this since most of our company's components are made with native-base. Thanks!

**Comment by dannyhw** [comment, marked as answer]
@SethY8s Hey thanks for trying out the beta, sorry that you ran into issues. Could you try updating your other storybook dependencies to match each other? Preferably remove any older packages on 5.3 you aren't using too so that you don't get clashing peer dependencies.

for example
```
 "@storybook/addon-actions": "^6.5",
  "@storybook/addon-ondevice-actions": "^6.0.1-beta.9",
``` 

I can take a look at your project in more detail tomorrow and debug it for you.

**Reply by SethY8s** [reply]
Good Morning @dannyhw,

Thank you so much I really appreciate the help! I just updated all my packages and am going to push my code to the repo again so you have the most up-to-date information. Unfortunately, I am still getting that strange HTML error with native-base. One thing I am going to try right now is using that prebuilt expo template and adding native-base to that right now and see if it is from one of my packages or just native-base.

**Reply by SethY8s** [reply]
@dannyhw 

Here's an update on what I found. I installed your expo react-native template [this one](https://github.com/dannyhw/expo-template-storybook) and got that running, however once I added native-base and made a story with one of their comments I was getting the same error "can't find variable HTMLElement". This makes me think this is a native-base X storybook v6 problem.

**Reply by dannyhw** [reply]
I'll look into it later tonight and see what I can find.

can you see where the error is coming from? any more information in the error? If you could send the full error that might be useful.

**Reply by SethY8s** [reply]
@dannyhw ,

Yes here is a picture of two errors and one warning. I checked and I get the same errors on both your template and my manual setup repo. Thank you again for helping!
![Screenshot 2022-11-10 123727](https://user-images.githubusercontent.com/95386742/201167803-1fd79df1-f52c-4da5-9f5b-5115296ad4b0.png)
![Screenshot 2022-11-10 123808](https://user-images.githubusercontent.com/95386742/201167822-acd848ec-5ef5-490d-bd5f-44da40ea3cec.png)
![Screenshot 2022-11-10 123823](https://user-images.githubusercontent.com/95386742/201167833-84a5cd2a-c4c2-4222-ab1a-89ef4d04f99a.png)

**Reply by dannyhw** [reply]
Yeah it seems like I can reproduce it, I think I know why though so let me investigate a solution and get back to you

**Reply by SethY8s** [reply]
Good Evening @dannyhw,

I know it is late on a Sunday so I don't expect a reply anytime soon, however, I was wondering if you have made any progress in this error? Or if not, maybe there was anything I could do to help resolve this problem since our company would love to use storybook v6, since it seems like v5 is more challenging to use. If you need anything or have any updates that would be really appreciated. Thanks!

-Seth

**Reply by dannyhw** [reply]
@SethY8s hey sorry for not getting back to you sooner, had a few busy days. I will debug it tonight.

**Reply by SethY8s** [reply]
@dannyhw, Excited to see what you come up with. Thanks for getting back to me so fast!

**Reply by dannyhw** [reply]
@SethY8s I was able to confirm my suspicions that this is caused by the polyfill added as a temporary workaround for some other issues in storybook/preview-web

I'm currently waiting for a new version of v6.5 that will allow me to remove this polyfill for good.

If you want to temporarily workaround this issue then there are some patches you can apply to node_modules for example using patch-package

in node_modules/@storybook/react-native/dist/index.js
remove or comment line 1 

```
// import './document-polyfill';
```

in node_modules/@storybook/preview-web/dist/modern/PreviewWeb.js
remove or comment line 28 and 29

```
    // this.view = new WebView();
    // this.urlStore = new UrlStore(); // Add deprecated APIs for back-compat
```

I've been told that the changes should be made next week so hopefully a permanent fix is coming very soon.

**Reply by jfdesgagne** [reply]
Hi, is there any news on this? I tried making the same patch changes but ran into more problem

> TypeError: undefined is not an object (evaluating 'document.location')

Thanks!

**Reply by jfdesgagne** [reply]
For now, applying the patch on `document-polyfill` to be:

```
// import Document from './DOM/Document';
// window.document = window.document || new Document();
window.document = {
    location: {

    },
    body: {
        addEventListener: () => undefined
    },
    addEventListener: () => undefined,
    querySelectorAll: () => [],
    getElementById: () => undefined
}

window.addEventListener = () => undefined
window.HTMLElement = {}
window.HTMLElement.prototype = {
    focus: () => undefined
}
```

get me around the problem for the moment

**Reply by dannyhw** [reply]
This is already fixed on the latest version of the beta, check the lock file to make sure you're on the latest @jfdesgagne
